lyncs_quda/gauge_field.py

```diff
@@ -367,11 +426,18 @@ def project(self, tol=None):
         is a destructive operation.  The number of link failures is
         reported so appropriate action can be taken.
         """
+        if self.location == "CPU":
+            raise NotImplementedError(
+                "This method currently works only when running on GPUs"
+            )
+
         if tol is None:
             tol = numpy.finfo(self.dtype).eps
+
+        assert self.device == cupy.cuda.runtime.getDevice()
         fails = cupy.zeros((1,), dtype="int32")
         lib.projectSU3(self.quda_field, tol, to_pointer(fails.data.ptr, "int *"))
-        return fails[0]
+        return fails.get()[0]  # shouldn't we reduce?
```

Comment:
"# shouldn't we reduce?"
It is already a scalar number. Remove comment if you agree